### PR TITLE
Add FirebaseAuth test_spec

### PR DESCRIPTION
**Pull request by paulb777**
- Add unit test spec for Auth.
- Add a complete Auth build and test rule.
- Testing is done via the pod gen generated workspace.
- AuthSample and its associated tests are migrated to their own isolated xcodeproj and associated Podfile.
- I've verified that CI will build and run the integration tests, but they're currently disabled in .travis.yml because of backend issues.
- Directory structure will be cleaned up in a future PR.
- Fix unit test warnings.
- Fix string in tests that failed in c99 builds.
- Eliminate sharedApplication mocks that expose https://github.com/erikdoe/ocmock/issues/346 on iPhone 4 test runs.
- The old build environment is still supported.
- It will be removed when all pods have transitioned to test_specs and pod gen.
- For now, see the instructions at https://github.com/firebase/firebase-ios-sdk/blob/master/Functions/README.md.
- pod gen FirebaseAuth.podspec --local-sources=./ --auto-open

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -43,6 +43,18 @@ jobs:
...
+    - stage: test
+      env:
+        - PROJECT=Auth PLATFORM=iOS METHOD=xcodebuild
+      before_install:
+        - ./scripts/if_changed.sh ./scripts/install_prereqs.sh
+      script:
+        - travis_retry ./scripts/if_changed.sh ./scripts/pod_lib_lint.rb FirebaseAuth.podspec --platforms=ios
+        - travis_retry ./scripts/if_changed.sh ./scripts/pod_lib_lint.rb FirebaseAuth.podspec --platforms=tvos
+        - travis_retry ./scripts/if_changed.sh ./scripts/pod_lib_lint.rb FirebaseAuth.podspec --platforms=macos
+        # TODO (renkelvin) Uncomment next line when Auth backend is fixed to support the integration tests.
```
Comment (renkelvin) [inline .travis.yml]: Got it.

File: .travis.yml [hunk 2]
```diff
@@ -161,11 +172,29 @@ jobs:
       before_install:
         - ./scripts/install_prereqs.sh
       script:
+        # TODO investigate why macos tests hang for FirebaseAuth - keychain related?
```
Comment (renkelvin) [inline .travis.yml]: I guess so. The OS asks for password when you want to access to the keychain.

File: Example/Auth/Tests/FIRAuthTests.m [hunk 3]
```diff
@@ -2350,23 +2353,25 @@ - (void)testAppDidReceiveRemoteNotificationWithCompletion_NotificationManagerHan
...
+  [self.fakeApplicationDelegate application:[GULAppDelegateSwizzler sharedApplication]
                didReceiveRemoteNotification:notification
                      fetchCompletionHandler:^(UIBackgroundFetchResult result) {}];
 
   [self.mockNotificationManager verify];
 }
 
 - (void)testAppOpenURL_AuthPresenterCanHandleURL {
-  NSURL *url = [NSURL URLWithString:@"https://localhost"];
+  if (@available(iOS 9.0, *)) {
```
Comment (renkelvin) [inline Example/Auth/Tests/FIRAuthTests.m]: Why no iOS 8.0?

File: .travis.yml (hunk 2 above)
Comment (paulb777) [reply .travis.yml]: I suspect that it's related to the password, but somehow the dynamic framework version works.

File: Example/Auth/Tests/FIRAuthTests.m (hunk 3 above)
Comment (paulb777) [reply Example/Auth/Tests/FIRAuthTests.m]: Added a comment.

**Pull request merged**